docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+Adds a dependency for the store or stores that map to the passed in string. If a store is passed adds a dependency for the passed in store. If an array is passed in adds a dependency for each element in the passed in array. Will do this for each input.
+
+```
+    storeOrInput = 'users'; //OR
+    storeOrInput = userStore; //OR
+    storeOrInput = ['users', roomStore]; //OR
+    storeOrInput = 'users'; additionalStoreOrInput = roomStore;
+
+```
+
+######<a id="itNeeds-architectureChain"></a>Return - `architectureChain` - type: OBJECT [#](#itNeeds-architectureChain)
+Returns this `architectureChain` instance to allow for continued method chaining.
+
+```
+  console.assert(architectureChain === architectureChain.itNeeds('users'));
+```
+
+#####<a id="architectureChain-itOutputs"></a>Method - `architectureChain.itOutputs` - type: FUNCTION [#](#architectureChain-itOutputs)
+Accepts any number of inputs: strings to output, or arrays of strings to output. Maps this `storeToArchitect` to the passed in output(s) so other stores can add dependencies on this store. If another store is already mapped to this output than stores that need the output will become dependent on both of these stores. Returns this `architectureChain` instance for continued method chaining.
+
+```
+    architectureChain.itOutputs(output, additionalOutputs...);
+```
+
+######<a id="itOutputs-output"></a>Parameter - `output` - type: STRING, ARRAY of STRINGs - required [#](#itOutputs-output)
+Registers the store to the passed in string such that other stores that add a need for the string will be dependent on this store. If an array is passed in registers each store in the passed in array. Will do this for each input. Multiple stores can have the same outputs, in which case a store that needs the string will be dependent on every store that outputs it.
+
+```
+    output = 'rooms'; //OR
+    output = ['rooms', 'unreadCount']; //OR
+    output = 'rooms'; additionalOutput = 'unreadCount';
+```
+
+######<a id="itOutputs-architectureChain"></a>Return - `architectureChain` - type: OBJECT [#](#itOutputs-architectureChain)
+Returns this `architectureChain` instance to allow for continued method chaining.
+
+```
+  console.assert(architectureChain === architectureChain.itOutputs('users'));
+```
+
+#####<a id="architectureChain-and"></a>Method - `architectureChain.and` - type: FUNCTION [#](#architectureChain-and)
+Clones the last invoked method for conjunctive method chaining. Thus, `and` will become the `itNeeds` method after `itNeeds` is invoked. After `itOuputs` is invoked `and` becomes the `itOutputs` method.
+
+```
+    architectureChain.itOutputs('rooms');
+    console.assert(architectureChain.and === architectureChain.itOutputs);
+
+    architectureChain.itNeeds('users');
+    console.assert(architectureChain.and === architectureChain.itNeeds);
+```
+
```

Comment:
Would it be beneficial to have a full example at the end of this doc? The examples throughout are great, so maybe it is a bit redundant to have a large example here. Maybe something that took full advantage of everything that is throughout the document?